tests/guardduty_test.py

```diff
@@ -11,90 +9,79 @@
-        self.assertTrue(gd_org_config['DataSources']['S3Logs']['AutoEnable'])
-
-    def test_guardduty_s3_protection_enabled_for_existing_accounts(self):
-        detector_management = guardduty.get_detector(DetectorId=(
-            guardduty.list_detectors()['DetectorIds'][0]))
-        self.assertEqual('ENABLED', detector_management['DataSources']['S3Logs']['Status'])
-
-        log_archive_account = self.control_tower_exection_role_session(self.get_log_archive_account_id())
-        guardduty_log_archive = log_archive_account.client('guardduty')
-        detector_log_archive = guardduty_log_archive.get_detector(DetectorId=(
-            guardduty_log_archive.list_detectors()['DetectorIds'][0]))
-
-        self.assertEqual('ENABLED', detector_log_archive['DataSources']['S3Logs']['Status'])
-
-    @classmethod
-    def control_tower_exection_role_session(cls, account_id):
-        account_creds = sts.assume_role(
-            RoleArn='arn:aws:iam::{}:role/AWSControlTowerExecution'.format(account_id),
-            RoleSessionName='superwerkertest'
-        )['Credentials']
-        return boto3.session.Session(
-            aws_access_key_id=account_creds['AccessKeyId'],
-            aws_secret_access_key=account_creds['SecretAccessKey'],
-            aws_session_token=account_creds['SessionToken']
-        )
+@pytest.fixture(scope="module")
+def management_account_id():
+    return sts.get_caller_identity()['Account']
+
+@pytest.fixture(scope="module")
+def audit_account_id():
+    return ssm.get_parameter(Name='/superwerker/account_id_audit')['Parameter']['Value']
+
+@pytest.fixture(scope="module")
+def log_archive_account_id():
+    return ssm.get_parameter(Name='/superwerker/account_id_logarchive')['Parameter']['Value']
+
+def control_tower_exection_role_session(account_id):
+    account_creds = sts.assume_role(
+        RoleArn='arn:aws:iam::{}:role/AWSControlTowerExecution'.format(account_id),
+        RoleSessionName='superwerkertest'
+    )['Credentials']
+    return boto3.session.Session(
+        aws_access_key_id=account_creds['AccessKeyId'],
+        aws_secret_access_key=account_creds['SecretAccessKey'],
+        aws_session_token=account_creds['SessionToken']
+    )
+
+def test_guardduty_enabled_with_delegated_admin_in_core_and_enrolled_accounts(audit_account_id, log_archive_account_id, management_account_id):
+    audit_account = control_tower_exection_role_session(account_id=audit_account_id)
+    guardduty_audit_account = audit_account.client('guardduty')
+
+    detector_id = guardduty_audit_account.list_detectors()['DetectorIds'][0]
+    members = guardduty_audit_account.list_members(DetectorId=detector_id, OnlyAssociated='true')['Members']
+
+    actual_members = [a['AccountId'] for a in members if a['RelationshipStatus'] == 'Enabled']
+
+    expected_members = [
+        log_archive_account_id,
+        management_account_id,
+    ]
+
+    assert expected_members == actual_members
```

Comment:
I think this might fail once not all accounts in the suspended OU are not actually suspended

Reply:
The accounts should be de-registered by the cleanup script, or not?

Reply:
I double-checked. It works because we use the manual invite/accept mechanism, not the organizational guard duty feature that automatically adds all accounts. We need to make sure that the clean-up script deletes the detectors properly since it still fails from time to time, so we avoid false positives.

Reply:
yes, generally aws is a bit slow in regards to integrating services on the organisational level. I'm pretty sure back then it was simply not available yet. the LZA for example uses the guard duty and security hub on org level